### PR TITLE
CMDCT-3444: [SAR] RE&T Field Validation

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Currently, the number fields in the SAR RE&T sections allow for any numbers, including decimals and negatives. As these represent people, they should _maaaaaybe_ not be negatives and decimals 😅

### Related ticket(s)

CMDCT-3444

---
### How to test

- Create and submit a WP as a state user (make sure to have applicable target populations, otherwise you won't be able to test the validation in the SAR)
- Approve that WP as an admin
- Create a SAR as a state user
- Navigate to the RE&T sections

You should get a validation error if you input a number less than one (1) or any decimals.

### Notes

N/A

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Comment by gmrabian** [comment]
I created and approved a WP on `stateuser1@test.com` in the deployed env. Feel free to jump straight to the SAR while validating :) 
https://de2379hujocus.cloudfront.net/

**Review by gmrabian — CHANGES_REQUESTED** [review]
Hi! Still need to fulfill two of the AC here:
- On blur, the value should be rounded to 0 places as needed.
- Fields in this section should allow whole numbers or N/A only, no decimals.

When I tested the fields errored on `N/A` and did not round decimals for me (which is how I read the first item above)

Screenshot 2024-04-22 at 10 00 38 AM

File: services/ui-src/src/utils/validation/schemas.ts [hunk 1]
```diff
@@ -82,6 +82,26 @@ export const validInteger = () =>
 export const validIntegerOptional = () =>
   validIntegerSchema().notRequired().nullable();
 
+// NUMBER NOT LESS THAN ONE
+export const numberNotLessThanOne = () =>
```
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/ui-src/src/utils/validation/schemas.ts]: That is correct, I just pushed an update for that! thank you ✨

File: services/ui-src/src/verbiage/errors.ts [hunk 2]
```diff
@@ -50,6 +50,7 @@ export const validationErrors = {
   NUMBER_LESS_THAN_ZERO: "Response must be greater than or equal to zero",
   INVALID_NUMBER: "Response must be a valid number",
   INVALID_NUMBER_OR_NA: 'Response must be a valid number or "N/A"',
+  INVALID_WHOLE_NUMBER: 'Response must be a valid whole number or "N/A"',
```
Comment (gmrabian) [inline services/ui-src/src/verbiage/errors.ts:53]: not used anywhere

**Review by gmrabian — DISMISSED** [review]
changes work as expected :)

**Review by ntsummers1 — COMMENTED** [review]
nit: I think it might be good to update the schemas.test.ts file with the new integer schema just to have our bases covered with it.

**Comment by karla-vm** [comment]
> nit: I think it might be good to update the schemas.test.ts file with the new integer schema just to have our bases covered with it.

Just updated the `app-api` test suite!

**Review by benmartin-coforma — COMMENTED** [review]
I have two comments:
1. I believe this removes the last reference to `checkStandardIntegerInputAgainstRegexes` in checkInputValidity.ts; if we're not using that, let's delete it.
2. One of the checks in that function validated the position of any dollar signs; without that check we can send the API dollar signs willy-nilly. The only restriction is that the first character must be a digit (because that's enough for `parseFloat` to be happy).

image

**Comment by jessabean** [comment]
Was about to approve this PR and then Ben commented about the dollar signs! Will 👀 after that fix. Let me know if you need any help!

**Comment by karla-vm** [comment]
> I have two comments:
> 
> 1. I believe this removes the last reference to `checkStandardIntegerInputAgainstRegexes` in checkInputValidity.ts; if we're not using that, let's delete it.
> 2. One of the checks in that function validated the position of any dollar signs; without that check we can send the API dollar signs willy-nilly. The only restriction is that the first character must be a digit (because that's enough for `parseFloat` to be happy).
> 
> image

Great catch, I brought back that check as part of my latest commit.

**Comment by karla-vm** [comment]
> Was about to approve this PR and then Ben commented about the dollar signs! Will 👀 after that fix. Let me know if you need any help!

it is ready for eyes 👀

**Review by benmartin-coforma — APPROVED** [review]
The code looks good!

**Pull request merged**